---
title: "RELIEF: methylation study"
author: "Mark Ziemann"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    fig_width: 7
    fig_height: 7
theme: cosmo
---

## Introduction


## Analysis pipeline

In this analysis, I will use the following resources:

* https://bioconductor.org/packages/release/bioc/vignettes/missMethyl/inst/doc/missMethyl.html

* https://www.bioconductor.org/help/course-materials/2015/BioC2015/methylation450k.html

* https://f1000research.com/articles/5-1281

This script should be run with R 4.0

```{r,packages}
suppressPackageStartupMessages({
    library("missMethyl")
    library("limma")
    library("minfi")
    library("IlluminaHumanMethylationEPICanno.ilm10b2.hg19")
    library("IlluminaHumanMethylationEPICmanifest")
    library("beeswarm")
    library("gplots")
    library("topconfects")
    library("FlowSorted.Blood.EPIC")
    library("mitch")
})
data(Locations)
source("epic_850k_functions.R")

```

# Obtaining array annotations

```{r,annotation}
anno <- getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)
myann <- data.frame(anno[,c("UCSC_RefGene_Name","Regulatory_Feature_Group")])
```
# Obtain reactome gene sets
```{r,genesets}
download.file("https://reactome.org/download/current/ReactomePathways.gmt.zip", 
    destfile="ReactomePathways.gmt.zip")
unzip("ReactomePathways.gmt.zip",overwrite = TRUE)
genesets <- gmt_import("ReactomePathways.gmt")
```
# Curate metadata

```{r,metadata}
tmp<-readLines("ILMLEPIC-15830_SampleSheet.csv")
sample.annotation<-read.csv(text=tmp[8:length(tmp)],stringsAsFactors=FALSE)
sample.annotation[grep("Pre",sample.annotation$Sample_Name),4]<-"NA"
sample.annotation[grep("T0",sample.annotation$Sample_Name),4]<-"Group1"
sample.annotation[grep("POD",sample.annotation$Sample_Name),4]<-"Group2"
sample.annotation$RG_number<-sapply(strsplit(sample.annotation$Sample_Name,"-"),"[",1)
# clinical metadata
samplesheet<-read.csv("samplesheet.tsv",sep="\t",stringsAsFactors=FALSE)
# merge these metadata together
sample.annotation2<-merge(sample.annotation,samplesheet,by="RG_number")
sample.annotation2$Basename<-sample.annotation2$Sentrix_ID
sample.annotation2$Basename<-paste(sample.annotation2$Sentrix_ID,sample.annotation2$Sentrix_Position,sep="_")
```

# Import, normalise and filter ####

```{r,methimport}
basedir = "analysis/idat"
rgSet <- read.metharray.exp(basedir, targets = sample.annotation2)

# normalise for differences in cell type composition
mSet<-preprocessRaw(rgSet)
mSetSw <- SWAN(mSet,verbose=FALSE)
# plots to show differences due to normalisation
par(mfrow=c(1,2), cex=1.25)
densityByProbeType(mSet[,1], main = "Raw")
densityByProbeType(mSetSw[,1], main = "SWAN")

detP <- detectionP(rgSet)
keep <- rowSums(detP < 0.01) == ncol(rgSet)
mSetSw <- mSetSw[keep,]

mset_reduced <- mSetSw
meth <- getMeth(mset_reduced)
unmeth <- getUnmeth(mset_reduced)
Mval <- log2((meth + 100)/(unmeth + 100))
beta <- getBeta(mset_reduced)
dim(Mval)

```


# plot mds
```{r,mds}
plotMDS(Mval, labels=sample.annotation2$Sample_Name, col=as.integer(factor(sample.annotation2$CrpGroup)))
legend("topleft",legend=c("CRP high","CRP norm"),pch=16,cex=1.2,col=1:2)
mds<-cmdscale(dist(t(Mval)))
rownames(mds)<-sample.annotation2$Sample_Name
g1<-rownames(mds[which(mds[,1]>0),])
g1<-unique(sapply(strsplit(g1,"-"),"[",1))
g0<-rownames(mds[which(mds[,1]<0),])
g0<-unique(sapply(strsplit(g0,"-"),"[",1))
# change header
Mval2<-Mval
colnames(Mval2)<-sample.annotation2$Sample_Name
```

# extract probes on chrX

```{r,sex prediction}
cgx<-rownames(Locations[which(Locations$chr %in% "chrX"),])
cgy<-rownames(Locations[which(Locations$chr %in% "chrY"),])
mvx<-Mval2[which(rownames(Mval2) %in% cgx),]
mvy<-Mval2[which(rownames(Mval2) %in% cgy),]
plot(colMeans(mvy),colMeans(mvx))
female <- names(which(colMeans(mvy) < -0))
female <- unique(sapply(strsplit(female,"-"),"[",1))
male <- names(which(colMeans(mvy) > 0))
male <- unique(sapply(strsplit(male,"-"),"[",1))
intersect(female,male)
# RG227 looks like a problem 
sample.annotation2$malemeth <- as.numeric(colMeans(mvy) > 0)
sample.annotation3 <- sample.annotation2[grep("RG227",sample.annotation2$Sample_Name,invert=TRUE),]
Mval3 <- Mval2[,grep("RG227",colnames(Mval2),invert=TRUE)]
write.table(sample.annotation3,file="samplesheet_mod.tsv",quote=FALSE,sep="\t")
save.image("missmethyl_epic_850k_analysis.Rdata")

```

# Differential analysis including sex chromosomes

## T0 vs POD analysis

Here we compare pre-op samples to post-op samples.

```{r,dm1}
name="t0vPod_in"
timepoint <- factor(as.numeric(grepl("POD",sample.annotation3$Sample_Name)))
age <- sample.annotation3$age
malemeth <- factor(sample.annotation3$malemeth)
design <- model.matrix(~age + malemeth + timepoint)
fit.reduced <- lmFit(Mval3,design)
fit.reduced <- eBayes(fit.reduced)
summary(decideTests(fit.reduced))
dm <- topTable(fit.reduced,coef=4, number = Inf)
dma <- merge(myann,dm,by=0)
dma <- dma[order(dma$P.Value),]
head(dma,10)
dm_up <- rownames(subset(dm,adj.P.Val<0.05 & logFC>0))
dm_dn <- rownames(subset(dm,adj.P.Val<0.05 & logFC<0))
length(dm_up)
length(dm_dn)
confects <- limma_confects(fit.reduced, coef=4, fdr=0.05)
make_dm_plots(dm = dm ,name=name , mx=Mval3, beta=beta, groups=timepoint, confects=confects)
t0vPod_in <- list("dma"=dma, "dm_up"=dm_up, "dm_dn"=dm_dn, "confects"=confects)
```

## Low vs High CRP samples post-op

Here we compare the post-op samples of patients with low CRP to high CRP.

```{r,dm2}
name="t1_crp_in"
sample.annotation3$timepoint <- factor(as.numeric(grepl("POD",sample.annotation3$Sample_Name)))
sample.annotation4 <- sample.annotation3[sample.annotation3$timepoint==1,]
Mval4 <- Mval3[,which(colnames(Mval3) %in% sample.annotation4$Sample_Name)]
crpgrp <- factor(sample.annotation4$CrpGroup,levels=c(0,1))
age <- sample.annotation4$age
malemeth <- factor(sample.annotation4$malemeth)
design <- model.matrix(~age + malemeth + crpgrp)
fit.reduced <- lmFit(Mval4,design)
fit.reduced <- eBayes(fit.reduced)
summary(decideTests(fit.reduced))
dm <- topTable(fit.reduced,coef=4, number = Inf)
dma <- merge(myann,dm,by=0)
dma <- dma[order(dma$P.Value),]
head(dma,10)
dm_up <- rownames(subset(dm,adj.P.Val<0.05 & logFC>0))
dm_dn <- rownames(subset(dm,adj.P.Val<0.05 & logFC<0))
length(dm_up)
length(dm_dn)
confects <- limma_confects(fit.reduced, coef=4, fdr=0.05)
make_dm_plots(dm = dm ,name=name , mx=Mval4, beta=beta, groups=crpgrp, confects=confects)
t0_crp_in <- list("dma"=dma, "dm_up"=dm_up, "dm_dn"=dm_dn, "confects"=confects)
```

## Low vs High CRP samples baseline

Here we compare the baseline samples of patients with low CRP to high CRP.

```{r,dm3}
name="t0_crp_in"
sample.annotation4 <- sample.annotation3[sample.annotation3$timepoint==0,]
Mval4 <- Mval3[,which(colnames(Mval3) %in% sample.annotation4$Sample_Name)]
crpgrp <- factor(sample.annotation4$CrpGroup,levels=c(0,1))
age <- sample.annotation4$age
malemeth <- factor(sample.annotation4$malemeth)
design <- model.matrix(~age + malemeth + crpgrp)
fit.reduced <- lmFit(Mval4,design)
fit.reduced <- eBayes(fit.reduced)
summary(decideTests(fit.reduced))
dm <- topTable(fit.reduced,coef=4, number = Inf)
dma <- merge(myann,dm,by=0)
dma <- dma[order(dma$P.Value),]
head(dma,10)
dm_up <- rownames(subset(dm,adj.P.Val<0.05 & logFC>0))
dm_dn <- rownames(subset(dm,adj.P.Val<0.05 & logFC<0))
length(dm_up)
length(dm_dn)
confects <- limma_confects(fit.reduced, coef=4, fdr=0.05)
make_dm_plots(dm = dm ,name=name , mx=Mval4, beta=beta, groups=crpgrp, confects=confects)
t0_crp_in <- list("dma"=dma, "dm_up"=dm_up, "dm_dn"=dm_dn, "confects"=confects)
```


# Differential analysis excluding sex chromosomes

## T0 vs POD analysis

Here we compare pre-op samples to post-op samples, excluding sex chromosomes.

```{r,dm4}
name="t0vPod_ex"
cgx<-rownames(Locations[which(Locations$chr %in% "chrX"),])
cgy<-rownames(Locations[which(Locations$chr %in% "chrY"),])
cgxy<-union(cgx,cgy)
Mval5 <- Mval3[which(!rownames(Mval3) %in% cgxy),]
timepoint <- factor(as.numeric(grepl("POD",sample.annotation3$Sample_Name)))
age <- sample.annotation3$age
malemeth <- factor(sample.annotation3$malemeth)
design <- model.matrix(~age + malemeth + timepoint)
fit.reduced <- lmFit(Mval5,design)
fit.reduced <- eBayes(fit.reduced)
summary(decideTests(fit.reduced))
dm <- topTable(fit.reduced,coef=4, number = Inf)
dma <- merge(myann,dm,by=0)
dma <- dma[order(dma$P.Value),]
head(dma,10)
dm_up <- rownames(subset(dm,adj.P.Val<0.05 & logFC>0))
dm_dn <- rownames(subset(dm,adj.P.Val<0.05 & logFC<0))
length(dm_up)
length(dm_dn)
confects <- limma_confects(fit.reduced, coef=4, fdr=0.05)
make_dm_plots(dm = dm ,name=name , mx=Mval5, beta=beta, groups=timepoint, confects=confects)
t0vPod_ex <- list("dma"=dma, "dm_up"=dm_up, "dm_dn"=dm_dn, "confects"=confects)

```

## Low vs High CRP samples post-op

Here we compare the post-op samples of patients with low CRP to high CRP excluding sex chromosomes.

```{r,dm5}
name="t1_crp_ex"
sample.annotation4 <- sample.annotation3[sample.annotation3$timepoint==1,]
Mval6 <- Mval3[,which(colnames(Mval3) %in% sample.annotation4$Sample_Name)]
Mval6 <- Mval6[which(!rownames(Mval6) %in% cgxy),]
crpgrp <- factor(sample.annotation4$CrpGroup,levels=c(0,1))
age <- sample.annotation4$age
malemeth <- factor(sample.annotation4$malemeth)
design <- model.matrix(~age + malemeth + crpgrp)
fit.reduced <- lmFit(Mval6,design)
fit.reduced <- eBayes(fit.reduced)
summary(decideTests(fit.reduced))
dm <- topTable(fit.reduced,coef=4, number = Inf)
dma <- merge(myann,dm,by=0)
dma <- dma[order(dma$P.Value),]
head(dma,10)
dm_up <- rownames(subset(dm,adj.P.Val<0.05 & logFC>0))
dm_dn <- rownames(subset(dm,adj.P.Val<0.05 & logFC<0))
length(dm_up)
length(dm_dn)
confects <- limma_confects(fit.reduced, coef=4, fdr=0.05)
make_dm_plots(dm = dm ,name=name , mx=Mval6, beta=beta, groups=crpgrp, confects=confects)
t0_crp_ex <- list("dma"=dma, "dm_up"=dm_up, "dm_dn"=dm_dn, "confects"=confects)
```

## Low vs High CRP samples

Here we compare the baseliine samples of patients with low CRP to high CRP excluding sex chromosomes.

```{r,dm6}
name="t0_crp_ex"
sample.annotation4 <- sample.annotation3[sample.annotation3$timepoint==0,]
Mval6 <- Mval3[,which(colnames(Mval3) %in% sample.annotation4$Sample_Name)]
Mval6 <- Mval6[which(!rownames(Mval6) %in% cgxy),]
crpgrp <- factor(sample.annotation4$CrpGroup,levels=c(0,1))
age <- sample.annotation4$age
malemeth <- factor(sample.annotation4$malemeth)
design <- model.matrix(~age + malemeth + crpgrp)
fit.reduced <- lmFit(Mval6,design)
fit.reduced <- eBayes(fit.reduced)
summary(decideTests(fit.reduced))
dm <- topTable(fit.reduced,coef=4, number = Inf)
dma <- merge(myann,dm,by=0)
dma <- dma[order(dma$P.Value),]
head(dma,10)
dm_up <- rownames(subset(dm,adj.P.Val<0.05 & logFC>0))
dm_dn <- rownames(subset(dm,adj.P.Val<0.05 & logFC<0))
length(dm_up)
length(dm_dn)
confects <- limma_confects(fit.reduced, coef=4, fdr=0.05)
make_dm_plots(dm = dm ,name=name , mx=Mval6, beta=beta, groups=crpgrp, confects=confects)
t0_crp_ex <- list("dma"=dma, "dm_up"=dm_up, "dm_dn"=dm_dn, "confects"=confects)
```

# Normalise for blood composition and repeat the analysis ####

```{r,methimport2}
basedir = "analysis/idat"
rgSet <- read.metharray.exp(basedir, targets = sample.annotation2)
cells <- estimateCellCounts2(rgSet,  referencePlatform= "IlluminaHumanMethylationEPIC", returnAll = TRUE)
mset <- cells$normalizedData

densityByProbeType(mSet[,1], main = "Raw")
densityByProbeType(mset[,1], main = "estimateCellCounts2")
mset_reduced <- mset[which(rownames(mset) %in% names(keep[keep==TRUE])),]
meth <- getMeth(mset_reduced)
unmeth <- getUnmeth(mset_reduced)
Mval <- log2((meth + 100)/(unmeth + 100))
beta <- getBeta(mset_reduced)
dim(Mval)

```


# plot mds
```{r,mds2}
plotMDS(Mval, labels=sample.annotation2$Sample_Name, col=as.integer(factor(sample.annotation2$CrpGroup)))
legend("topleft",legend=c("CRP high","CRP norm"),pch=16,cex=1.2,col=1:2)
mds<-cmdscale(dist(t(Mval)))
rownames(mds)<-sample.annotation2$Sample_Name
g1<-rownames(mds[which(mds[,1]>0),])
g1<-unique(sapply(strsplit(g1,"-"),"[",1))
g0<-rownames(mds[which(mds[,1]<0),])
g0<-unique(sapply(strsplit(g0,"-"),"[",1))
# change header
Mval2<-Mval
colnames(Mval2)<-sample.annotation2$Sample_Name
```

# Differential analysis including sex chromosomes

## T0 vs POD analysis

Here we compare pre-op samples to post-op samples.

```{r,dm7}
name="t0vPod_in_blood"
timepoint <- factor(as.numeric(grepl("POD",sample.annotation3$Sample_Name)))
age <- sample.annotation3$age
malemeth <- factor(sample.annotation3$malemeth)
design <- model.matrix(~age + malemeth + timepoint)
fit.reduced <- lmFit(Mval3,design)
fit.reduced <- eBayes(fit.reduced)
summary(decideTests(fit.reduced))
dm <- topTable(fit.reduced,coef=4, number = Inf)
dma <- merge(myann,dm,by=0)
dma <- dma[order(dma$P.Value),]
head(dma,10)
dm_up <- rownames(subset(dm,adj.P.Val<0.05 & logFC>0))
dm_dn <- rownames(subset(dm,adj.P.Val<0.05 & logFC<0))
length(dm_up)
length(dm_dn)
confects <- limma_confects(fit.reduced, coef=4, fdr=0.05)
make_dm_plots(dm = dm ,name=name , mx=Mval3, beta=beta, groups=timepoint, confects=confects)
t0vPod_in_blood <- list("dma"=dma, "dm_up"=dm_up, "dm_dn"=dm_dn, "confects"=confects)
```

## Low vs High CRP samples post-op

Here we compare the post-op samples of patients with low CRP to high CRP.

```{r,dm8}
name="t1_crp_in_blood"
sample.annotation4 <- sample.annotation3[sample.annotation3$timepoint==1,]
Mval4 <- Mval3[,which(colnames(Mval3) %in% sample.annotation4$Sample_Name)]
crpgrp <- factor(sample.annotation4$CrpGroup,levels=c(0,1))
age <- sample.annotation4$age
malemeth <- factor(sample.annotation4$malemeth)
design <- model.matrix(~age + malemeth + crpgrp)
fit.reduced <- lmFit(Mval4,design)
fit.reduced <- eBayes(fit.reduced)
summary(decideTests(fit.reduced))
dm <- topTable(fit.reduced,coef=4, number = Inf)
dma <- merge(myann,dm,by=0)
dma <- dma[order(dma$P.Value),]
head(dma,10)
dm_up <- rownames(subset(dm,adj.P.Val<0.05 & logFC>0))
dm_dn <- rownames(subset(dm,adj.P.Val<0.05 & logFC<0))
length(dm_up)
length(dm_dn)
confects <- limma_confects(fit.reduced, coef=4, fdr=0.05)
make_dm_plots(dm = dm ,name=name , mx=Mval4, beta=beta, groups=crpgrp, confects=confects)
t0_crp_in_blood <- list("dma"=dma, "dm_up"=dm_up, "dm_dn"=dm_dn, "confects"=confects)
```


## Low vs High CRP samples baseline

Here we compare the baseline samples of patients with low CRP to high CRP.

```{r,dm9}
name="t0_crp_in_blood"
sample.annotation4 <- sample.annotation3[sample.annotation3$timepoint==0,]
Mval4 <- Mval3[,which(colnames(Mval3) %in% sample.annotation4$Sample_Name)]
crpgrp <- factor(sample.annotation4$CrpGroup,levels=c(0,1))
age <- sample.annotation4$age
malemeth <- factor(sample.annotation4$malemeth)
design <- model.matrix(~age + malemeth + crpgrp)
fit.reduced <- lmFit(Mval4,design)
fit.reduced <- eBayes(fit.reduced)
summary(decideTests(fit.reduced))
dm <- topTable(fit.reduced,coef=4, number = Inf)
dma <- merge(myann,dm,by=0)
dma <- dma[order(dma$P.Value),]
head(dma,10)
dm_up <- rownames(subset(dm,adj.P.Val<0.05 & logFC>0))
dm_dn <- rownames(subset(dm,adj.P.Val<0.05 & logFC<0))
length(dm_up)
length(dm_dn)
confects <- limma_confects(fit.reduced, coef=4, fdr=0.05)
make_dm_plots(dm = dm ,name=name , mx=Mval4, beta=beta, groups=crpgrp, confects=confects)
t0_crp_in_blood <- list("dma"=dma, "dm_up"=dm_up, "dm_dn"=dm_dn, "confects"=confects)
```

# Differential analysis excluding sex chromosomes

## T0 vs POD analysis

Here we compare pre-op samples to post-op samples, excluding X and Y chromosomes.

```{r,dm10}
name="t0vPod_ex_blood"
cgx<-rownames(Locations[which(Locations$chr %in% "chrX"),])
cgy<-rownames(Locations[which(Locations$chr %in% "chrY"),])
cgxy<-union(cgx,cgy)
Mval5 <- Mval3[which(!rownames(Mval3) %in% cgxy),]
timepoint <- factor(as.numeric(grepl("POD",sample.annotation3$Sample_Name)))
age <- sample.annotation3$age
malemeth <- factor(sample.annotation3$malemeth)
design <- model.matrix(~age + malemeth + timepoint)
fit.reduced <- lmFit(Mval5,design)
fit.reduced <- eBayes(fit.reduced)
summary(decideTests(fit.reduced))
dm <- topTable(fit.reduced,coef=4, number = Inf)
dma <- merge(myann,dm,by=0)
dma <- dma[order(dma$P.Value),]
head(dma,10)
dm_up <- rownames(subset(dm,adj.P.Val<0.05 & logFC>0))
dm_dn <- rownames(subset(dm,adj.P.Val<0.05 & logFC<0))
length(dm_up)
length(dm_dn)
confects <- limma_confects(fit.reduced, coef=4, fdr=0.05)
make_dm_plots(dm = dm ,name=name , mx=Mval5, beta=beta, groups=timepoint, confects=confects)
t0vPod_ex_blood <- list("dma"=dma, "dm_up"=dm_up, "dm_dn"=dm_dn, "confects"=confects)

# pathway
dmap <- dma[grep("Promoter_Associated",dma$Regulatory_Feature_Group),]
dmap[which(dmap$UCSC_RefGene_Name==""),2] <- "NA"
dmap$genename <- sapply(strsplit(dmap$UCSC_RefGene_Name,";"),"[[",1)
dmap2 <- dmap[,c("genename","t")]
rank <- aggregate(. ~ genename,dmap2,mean)
rownames(rank) <- rank$genename
rank$genename=NULL
write.table(rank,file="t0vPod_meth.tsv",sep="\t",quote=FALSE)
capture.output(
    res <- mitch_calc(x = rank,genesets = genesets, priority = "significance",resrows=20)
    , file = "/dev/null", append = FALSE,
    type = c("output", "message"), split = FALSE)
head(res$enrichment_result,20)
unlink("t0vPod_ex_blood.html")
capture.output(
    mitch_plots(res,outfile="t0vPod_ex_blood.pdf")
    , file = "/dev/null", append = FALSE,
    type = c("output", "message"), split = FALSE)

```


## Low vs High CRP samples post-op

Here we compare the post-op samples of patients with low CRP to high CRP excluding sex chromosomes.

```{r,dm11}
name="t1_crp_ex_blood"
sample.annotation4 <- sample.annotation3[sample.annotation3$timepoint==1,]
Mval6 <- Mval3[,which(colnames(Mval3) %in% sample.annotation4$Sample_Name)]
Mval6 <- Mval6[which(!rownames(Mval6) %in% cgxy),]
crpgrp <- factor(sample.annotation4$CrpGroup,levels=c(0,1))
age <- sample.annotation4$age
malemeth <- factor(sample.annotation4$malemeth)
design <- model.matrix(~age + malemeth + crpgrp)
fit.reduced <- lmFit(Mval6,design)
fit.reduced <- eBayes(fit.reduced)
summary(decideTests(fit.reduced))
dm <- topTable(fit.reduced,coef=4, number = Inf)
dma <- merge(myann,dm,by=0)
dma <- dma[order(dma$P.Value),]
head(dma,10)
dm_up <- rownames(subset(dm,adj.P.Val<0.05 & logFC>0))
dm_dn <- rownames(subset(dm,adj.P.Val<0.05 & logFC<0))
length(dm_up)
length(dm_dn)
confects <- limma_confects(fit.reduced, coef=4, fdr=0.05)
make_dm_plots(dm = dm ,name=name , mx=Mval6, beta=beta, groups=crpgrp, confects=confects)
t1_crp_ex_blood <- list("dma"=dma, "dm_up"=dm_up, "dm_dn"=dm_dn, "confects"=confects)

# pathway
dmap <- dma[grep("Promoter_Associated",dma$Regulatory_Feature_Group),]
dmap[which(dmap$UCSC_RefGene_Name==""),2] <- "NA"
dmap$genename <- sapply(strsplit(dmap$UCSC_RefGene_Name,";"),"[[",1)
dmap2 <- dmap[,c("genename","t")]
rank <- aggregate(. ~ genename,dmap2,mean)
rownames(rank) <- rank$genename
rank$genename=NULL
write.table(rank,file="t1_crp_meth.tsv",sep="\t",quote=FALSE)
capture.output( 
    res <- mitch_calc(x = rank,genesets = genesets, priority = "significance",resrows=20)
    , file = "/dev/null", append = FALSE,
    type = c("output", "message"), split = FALSE)
head(res$enrichment_result,20)
unlink("t1_crp_ex_blood.html")
capture.output( 
    mitch_plots(res,outfile="t1_crp_ex_blood.pdf")
    , file = "/dev/null", append = FALSE,
    type = c("output", "message"), split = FALSE)
```

## Low vs High CRP samples

Here we compare the baseline samples of patients with low CRP to high CRP excluding sex chromosomes.

```{r,dm12}
name="t0_crp_ex_blood"
sample.annotation4 <- sample.annotation3[sample.annotation3$timepoint==0,]
Mval6 <- Mval3[,which(colnames(Mval3) %in% sample.annotation4$Sample_Name)]
Mval6 <- Mval6[which(!rownames(Mval6) %in% cgxy),]
crpgrp <- factor(sample.annotation4$CrpGroup,levels=c(0,1))
age <- sample.annotation4$age
malemeth <- factor(sample.annotation4$malemeth)
design <- model.matrix(~age + malemeth + crpgrp)
fit.reduced <- lmFit(Mval6,design)
fit.reduced <- eBayes(fit.reduced)
summary(decideTests(fit.reduced))
dm <- topTable(fit.reduced,coef=4, number = Inf)
dma <- merge(myann,dm,by=0)
dma <- dma[order(dma$P.Value),]
head(dma,10)
dm_up <- rownames(subset(dm,adj.P.Val<0.05 & logFC>0))
dm_dn <- rownames(subset(dm,adj.P.Val<0.05 & logFC<0))
length(dm_up)
length(dm_dn)
confects <- limma_confects(fit.reduced, coef=4, fdr=0.05)
make_dm_plots(dm = dm ,name=name , mx=Mval6, beta=beta, groups=crpgrp, confects=confects)
t0_crp_ex_blood <- list("dma"=dma, "dm_up"=dm_up, "dm_dn"=dm_dn, "confects"=confects)

# pathway
dmap <- dma[grep("Promoter_Associated",dma$Regulatory_Feature_Group),]
dmap[which(dmap$UCSC_RefGene_Name==""),2] <- "NA"
dmap$genename <- sapply(strsplit(dmap$UCSC_RefGene_Name,";"),"[[",1)
dmap2 <- dmap[,c("genename","t")]
rank <- aggregate(. ~ genename,dmap2,mean)
rownames(rank) <- rank$genename
rank$genename=NULL
write.table(rank,file="t0_crp_meth.tsv",sep="\t",quote=FALSE)
capture.output( 
    res <- mitch_calc(x = rank,genesets = genesets, priority = "significance",resrows=20)
    , file = "/dev/null", append = FALSE,
    type = c("output", "message"), split = FALSE)
head(res$enrichment_result,20)
unlink("t0_crp_ex_blood.html")
capture.output(
    mitch_plots(res,outfile="t0_crp_ex_blood.pdf")
    , file = "/dev/null", append = FALSE,
    type = c("output", "message"), split = FALSE)

```
